### PR TITLE
 example(tracing): Add tracing example

### PR DESCRIPTION
**Pull request by yejiayu**
**What this PR does / why we need it**:
Add tracing example

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ddysher** [comment]
@yejiayu can you provide a howto guide?

**Comment by ddysher** [comment]
Does this depends on https://github.com/caicloud/nirvana/pull/116?

**Comment by yejiayu** [comment]
They are independent

**Comment by kdada** [comment]
The package `github.com/opentracing-contrib/go-stdlib/nethttp` is not good. Update #116 first.

**Comment by yejiayu** [comment]
ping @ddysher can you merge it?

File: examples/tracing-basic/main.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package main
+
+import (
+	"context"
+	"errors"
+
+	"github.com/caicloud/nirvana/config"
+	"github.com/caicloud/nirvana/definition"
+	"github.com/caicloud/nirvana/log"
+	tracingMid "github.com/caicloud/nirvana/plugins/tracing/middleware"
```
Comment (kdada) [inline examples/tracing-basic/main.go]: Why is this plugin placed at `github.com/caicloud/nirvana/plugins/tracing/middleware`

**Pull request merged**